lab12_python.py

```diff
@@ -0,0 +1,65 @@
+import os
+import hashlib
+import threading
+
+def duplicates(folder):
+    dup = {}
+    for dirname, root, files in os.walk(folder):
+        print(f'Scanning {dirname}...')
+        for filename in files:
+            path = os.path.join(dirname, filename)
+            hash = get_hash(path)
+            if hash in dup:
+                dup[hash].append(path)
+            else:
+                dup[hash] = [path]
+    return dup
+
+def get_hash(path, blocksize = 65000):
+    with open(path, 'rb') as file:
+        hash = hashlib.md5()
+        buf = file.read(blocksize)
+        if len(buf) > 1:
+            hash.update(buf)
+            buf = file.read(blocksize)
+        hasher = hash.digest()
+    return hasher
+
+def joinDicts(dict1, dict2):
```

Comment:
python dictionary has an embedded method for joining dicts, do not reinvent the wheel